src/ensembl/src/content/app/entity-viewer/state/general/entityViewerGeneralActions.ts

```diff
@@ -45,93 +45,90 @@ export const setActiveGenomeId = createAction(
   'entity-viewer/set-active-genome-id'
 )<string>();
 
-export const setDataFromUrl = (
-  params: EntityViewerParams
-): ThunkAction<void, any, null, Action<string>> => (
-  dispatch,
-  getState: () => RootState
-) => {
-  const state = getState();
-  const { genomeId: genomeIdFromUrl } = params;
-
-  let activeGenomeId = getEntityViewerActiveGenomeId(state);
-  const activeEntityId = getEntityViewerActiveEntityId(state) || undefined;
-
-  const entityIdForUrl = activeEntityId
-    ? buildFocusIdForUrl(activeEntityId)
-    : undefined;
-
-  if (!genomeIdFromUrl && !activeGenomeId) {
-    dispatch(setDefaultActiveGenomeId());
-    activeGenomeId = getEntityViewerActiveGenomeId(state) as string;
-  } else if (!genomeIdFromUrl && activeGenomeId) {
-    const newUrl = urlHelper.entityViewer({
-      genomeId: activeGenomeId,
-      entityId: entityIdForUrl
-    });
-    dispatch(replace(newUrl));
-  } else if (genomeIdFromUrl && genomeIdFromUrl !== activeGenomeId) {
-    dispatch(setActiveGenomeId(genomeIdFromUrl));
-    dispatch(fetchGenomeData(genomeIdFromUrl));
-    dispatch(ensureSpeciesIsEnabled(genomeIdFromUrl));
-
-    // TODO: when backend is ready, entity info may also need fetching
-  } else if (activeGenomeId) {
-    // TODO: when backend is ready, fetch entity info
-    const genomeInfo = getGenomeInfoById(state, activeGenomeId);
-    if (!genomeInfo) {
-      dispatch(fetchGenomeData(activeGenomeId));
+export const setDataFromUrl =
+  (params: EntityViewerParams): ThunkAction<void, any, null, Action<string>> =>
+  (dispatch, getState: () => RootState) => {
+    const state = getState();
+    const { genomeId: genomeIdFromUrl } = params;
+
+    let activeGenomeId = getEntityViewerActiveGenomeId(state);
+    const activeEntityId = getEntityViewerActiveEntityId(state) || undefined;
+
+    const entityIdForUrl = activeEntityId
```

Comment:
Same question as above: why `null` instead of `undefined`? Think we will need to take a look at the type `EntityViewerUrlParams` for this.